README.md

```diff
@@ -52,6 +51,74 @@ To reload the extension after some changes, open [`chrome://extensions`](chrome:
+
+During initialization, the entrypoint script also sets up an array of `Page` objects. Each `Page` represents a specific type of page on the website where our widget can load. They handle all the site-specific logic. For instance, to confirm we're on a particular type of page, a `Page` object would use the `#detect()` method. Or, to figure out where to place our widget, it would use `#findInsertionPoint()`. You can get more insights into these functions in the superclass at `src/pages/page.js`.
+
+The entry point's final task is to kickstart the `src/widget/widgetOrchestrator.js`. This script checks if our widget can load on the current page and inserts a button, the widget, and a slot for additional data in appropriate zones on the supported sites.
+
+Lastly, most of the remaining logic takes place in `src/components/widget.js`, which handles interactions with the Orbit API and manages the widget's various display states.
+
+## Adding Support for a New Site
+
+**For an example of this process being followed: https://github.com/orbit-love/orbit-browser-extension/pull/45**
+
+To add support for a new site to the extension:
+
+1. Create a New Entrypoint
+
+Start by creating a new entry point file, similar to src/widget/githubEntrypoint.js. This will require understanding when to load the widget for this specific site. Broadly speaking, `DOMContentLoaded` is a good place to start. However different sites have different requirements, so it will require some investigation to identify the appropriate events to observe for the page load lifecycle.
+
+In creating this, you'll reach a point where you need to define the pages to pass to the widget orchestrator. So...
+
+2. Create a New Page Class
+
+Create a new Page class for the new site by extending `src/pages/page.js`. This class will handles page-specific logic - see the comments in the superclass for details on the functions to implement, and take inspiration from the existing types of page.
+
+3. Create a Page-Specific Button
+
+This is a custom element that needs to match the styling of the new site. This will be the button that users click to open the widget. By default the widget should be named `obe-#{platform}-button` where platform is an identifier for the new site (ie "github"), but you can customise this by overriding the getButtonElementName function in your page.
+
+4. Modify the Manifest.json File
+
+Update src/manifest.json to specify on which URLs the new entry point script should run. Be sure to follow the correct JSON syntax and the Google Chrome extension manifest version specifications.
+
+5. (Optional) Supply AdditionalData in Widget
+
+The widget contains a slot called additionalData, which you can update with site-specific content. For example, on Github we show the number of github contributions the member has made.This may require some more involved engineering to make work, including adding new requests to the background runner.
```

Comment:
- Capitalization: “GitHub”
- I think there’s a missing space before `This may require`…

I’d also suggest dropping the “This may require some more involved…” sentence, as it doesn’t add much. Either we remove it (fine with me), or we explain a bit more how it works. However this is a bit of an experimental bit at the moment since it only is available for GitHub, so I wouldn’t venture too far on technical documentation just yet